Setup/InstallSchema.php

```diff
@@ -174,4 +174,95 @@ protected function installAvataxTaxClassOpColumn(SchemaSetupInterface $setup, Mo
                 );
         }
     }
+
+    /**
+     * Install avatax_queue table
+     *
+     * @param \Magento\Framework\Setup\SchemaSetupInterface   $setup
+     * @param \Magento\Framework\Setup\ModuleContextInterface $context
+     * @return $this
+     */
+    protected function installAvataxQueueTable(SchemaSetupInterface $setup, ModuleContextInterface $context)
+    {
+        $table = $setup->getConnection()->newTable($setup->getTable('avatax_queue'))
+            ->addColumn(
+                'queue_id',
+                Table::TYPE_INTEGER,
+                null,
+                ['identity' => true, 'unsigned' => true, 'nullable' => false, 'primary' => true],
+                'Queue ID'
+            )->addColumn(
+                'store_id',
+                Table::TYPE_SMALLINT,
+                null,
+                ['unsigned' => true, 'default' => '0'],
+                'Store Id'
+            )->addColumn(
+                'entity_id',
+                Table::TYPE_INTEGER,
+                null,
+                ['unsigned' => true, 'nullable' => false],
+                'Entity id'
+            )->addColumn(
+                'entity_increment_id',
+                Table::TYPE_TEXT,
+                50,
+                [],
+                'Entity increment id'
+            )->addColumn(
+                'type',
+                Table::TYPE_TEXT,
+                50,
+                [],
+                'Type'
+            )->addColumn(
+                'status',
+                Table::TYPE_TEXT,
+                50,
+                [],
+                'Status'
+            )->addColumn(
+                'attempt',
+                Table::TYPE_SMALLINT,
+                null,
+                ['unsigned' => true, 'default' => '0'],
+                'Attempt'
+            )->addColumn(
+                'message',
+                Table::TYPE_TEXT,
+                null,
+                [],
+                'Message'
+            )->addColumn(
+                'created_at',
+                Table::TYPE_TIMESTAMP,
+                null,
+                ['nullable' => false, 'default' => Table::TIMESTAMP_INIT],
+                'Created At'
+            )->addColumn(
+                'updated_at',
+                Table::TYPE_TIMESTAMP,
+                null,
+                ['nullable' => false, 'default' => Table::TIMESTAMP_INIT],
+                'Updated At'
+            )->addForeignKey(
+                $setup->getFkName('avatax_queue', 'store_id', 'store', 'store_id'),
+                'store_id',
+                $setup->getTable('store'),
+                'store_id',
+                Table::ACTION_CASCADE
+            )->addIndex(
```

Comment:
It's not correct to have one combined index on many fields. Please do similar we have in avatax for Magento1 version.